src/network/networkd-state-file.c

```diff
@@ -136,6 +137,20 @@ int manager_save(Manager *m) {
                 if (!link->network)
                         continue;
 
+                if (link->network->required_for_online) {
+                        if (online_state == LINK_ONLINE_STATE_PARTIALLY_ONLINE) {
+                                /* nothing to do - the state cannot change from here */
+                        } else if (online_state == LINK_ONLINE_STATE_ONLINE &&
+                                   link->online_state != LINK_ONLINE_STATE_ONLINE) {
+                                /* not all required links are online - mark as partially online */
+                                online_state = LINK_ONLINE_STATE_PARTIALLY_ONLINE;
+                        }
+                        else if (online_state < link->online_state) {
+                                /* push up the online state */
+                                online_state = link->online_state;
+                        }
```

Comment:
coding style, plesae place `else if` on same line as closing `}`. And no {} around single line if blocks.

https://systemd.io/CODING_STYLE/#formatting

Reply:
This original implementation was hard to read and had a bug, so I went for an alternative solution. Please check.